pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java

```diff
@@ -266,6 +271,8 @@ public ProducerImpl(PulsarClientImpl client, String topic, ProducerConfiguration
             }
             this.batchMessageContainer = (BatchMessageContainerBase) containerBuilder.build();
             this.batchMessageContainer.setProducer(this);
+            ThreadFactory threadFactory = new ExecutorProvider.ExtendedThreadFactory("pulsar-batch", Thread.currentThread().isDaemon());
```

Comment:
Can we reuse an existing thread pool? Otherwise each ProducerImpl(partition) will create a thread pool

Reply:
resue the scheduleExecutorProvider of PulsarClientImpl